File: src/main/java/eladkay/quaritum/common/item/soulstones/ItemPassionateSoulstone.java
package eladkay.quaritum.common.item.soulstones;

import eladkay.quaritum.api.animus.AnimusHelper;
import eladkay.quaritum.api.animus.INetworkProvider;
import eladkay.quaritum.common.item.base.ItemMod;
import eladkay.quaritum.common.lib.LibNames;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;
import net.minecraft.world.World;
import net.minecraftforge.fml.common.IFuelHandler;
import net.minecraftforge.fml.common.registry.GameRegistry;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;
import java.util.List;

public class ItemPassionateSoulstone extends ItemMod implements INetworkProvider, IFuelHandler {

    public ItemPassionateSoulstone() {
        super(LibNames.PASSIONATE_SOULSTONE);
        setMaxStackSize(1);
        GameRegistry.registerFuelHandler(this);
    }

    @Override
    public ItemStack getContainerItem(@Nonnull ItemStack itemStack) {
        if (getPlayer(itemStack) == null) return null;
        AnimusHelper.Network.addAnimus(getPlayer(itemStack), -4);
        ItemStack copiedStack = itemStack.copy();
        copiedStack.stackSize = 1;
        return copiedStack;
    }

    @Override
    public void addInformation(ItemStack stack, EntityPlayer playerIn, List<String> tooltip, boolean advanced) {
        AnimusHelper.Network.addInformation(stack, tooltip, advanced);
    }

    @Override
    public boolean hasContainerItem(ItemStack itemStack) {
        return true;
    }

    @Override
    public boolean isProvider(@Nonnull ItemStack stack) {
        return false;
    }

    @Override
    public boolean isReceiver(@Nonnull ItemStack stack) {
        return false;
    }

    @Override
    public int getEntityLifespan(ItemStack itemStack, World world) {
        return Integer.MAX_VALUE;
    }

    @Override
    public int getBurnTime(ItemStack fuel) {
        return fuel.getItem() == this && AnimusHelper.Network.getAnimus(getPlayer(fuel)) >= 4 ? 200 : 0;
    }
}

File: src/main/java/eladkay/quaeritum/common/item/soulstones/ItemPassionateSoulstone.kt
package eladkay.quaeritum.common.item.soulstones

import com.teamwizardry.librarianlib.features.base.item.ItemMod
import eladkay.quaeritum.api.animus.AnimusHelper
import eladkay.quaeritum.api.animus.EnumAnimusTier
import eladkay.quaeritum.api.animus.INetworkProvider
import eladkay.quaeritum.common.lib.LibNames
import net.minecraft.client.util.ITooltipFlag
import net.minecraft.entity.Entity
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.init.SoundEvents
import net.minecraft.item.ItemStack
import net.minecraft.util.ActionResult
import net.minecraft.util.EnumActionResult
import net.minecraft.util.EnumHand
import net.minecraft.util.SoundCategory
import net.minecraft.world.World
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly

class ItemPassionateSoulstone : ItemMod(LibNames.PASSIONATE_SOULSTONE), INetworkProvider {

    init {
        setMaxStackSize(1)
    }

    override fun getContainerItem(itemStack: ItemStack): ItemStack {
        itemStack.grow(1)
        val copiedStack = itemStack.copy()
        itemStack.shrink(1)
        if (!AnimusHelper.requestAnimus(copiedStack, 4, EnumAnimusTier.LUCIS, true)) return ItemStack.EMPTY
        return copiedStack
    }

    @SideOnly(Side.CLIENT)
    override fun addInformation(stack: ItemStack, world: World?, tooltip: MutableList<String>, advanced: ITooltipFlag) {
        AnimusHelper.Network.addInformation(stack, tooltip)
    }

    override fun hasContainerItem(itemStack: ItemStack?): Boolean {
        return true
    }

    override fun isProvider(stack: ItemStack): Boolean {
        return false
    }

    override fun isReceiver(stack: ItemStack): Boolean {
        return false
    }

    override fun getEntityLifespan(itemStack: ItemStack?, world: World?): Int {
        return Integer.MAX_VALUE
    }

    override fun onUpdate(stack: ItemStack, worldIn: World?, entityIn: Entity, itemSlot: Int, isSelected: Boolean) {
        if (getPlayer(stack) == null && entityIn is EntityPlayer)
            setPlayer(stack, entityIn.uniqueID)
    }

    override fun getItemBurnTime(fuel: ItemStack): Int {
        return if (AnimusHelper.hasMinimum(fuel, 4, EnumAnimusTier.LUCIS)) 200 else 0
    }

    override fun onItemRightClick(worldIn: World, playerIn: EntityPlayer, hand: EnumHand): ActionResult<ItemStack> {
        val itemStackIn = playerIn.getHeldItem(hand)

        if (playerIn.isSneaking) {
            setPlayer(itemStackIn, playerIn.uniqueID)
            worldIn.playSound(playerIn, playerIn.position, SoundEvents.ITEM_ARMOR_EQUIP_IRON, SoundCategory.PLAYERS, 1f, 1f)
            return ActionResult(EnumActionResult.SUCCESS, itemStackIn)
        }
        return super.onItemRightClick(worldIn, playerIn, hand)
    }
}

